src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -653,36 +572,30 @@ internal virtual void OutputProperties(ArrayList list)
         /// </summary>
         internal void WriteItems(SortedList itemTypes)
         {
-            if (Verbosity == LoggerVerbosity.Diagnostic && showItemAndPropertyList)
-            {
-                if (itemTypes.Count == 0)
-                {
-                    return;
-                }
+            if (Verbosity != LoggerVerbosity.Diagnostic || !showItemAndPropertyList || itemTypes.Count == 0) return;
 
-                // Write the banner
-                setColor(ConsoleColor.Green);
-                WriteLinePretty(currentIndentLevel, ResourceUtilities.FormatResourceString("ItemListHeader"));
-
-                // Write each item type and its itemspec, one per line
-                foreach (DictionaryEntry entry in itemTypes)
-                {
-                    string itemType = (string)entry.Key;
-                    ArrayList itemTypeList = (ArrayList)entry.Value;
+            // Write the banner
+            setColor(ConsoleColor.Green);
+            WriteLinePretty(currentIndentLevel, ResourceUtilities.FormatResourceString("ItemListHeader"));
 
-                    if (itemTypeList.Count == 0)
-                    {
-                        continue;
-                    }
+            // Write each item type and its itemspec, one per line
+            foreach (DictionaryEntry entry in itemTypes)
+            {
+                string itemType = (string) entry.Key;
+                ArrayList itemTypeList = (ArrayList) entry.Value;
```

Comment:
We don't normally have spaces between the cast and the variable, do we? Surprised this came out this way.